x-pack/plugins/infra/server/plugin.ts

```diff
@@ -259,21 +264,26 @@ export class InfraServerPlugin
       ]);
     }
 
-    initInfraServer(this.libs);
```

Comment:
You mentioned
> "Having endpoint initialization in start makes it more convenient to access start dependencies which almost all endpoints require." 

What kind of start dependencies do most of the endpoints require?

Reply:
Depends on the endpoint, but there is a `getStartServices` that we put into `libs` during `start` phase and pass down to route initializers. [Here are instances](https://github.com/search?q=repo:elastic/kibana+path:x-pack/plugins/infra/server/+getStartServices()&type=code) of where this method is used. Creating this loophole from the `start` phase in form of `getStartServices()` feels a bit weird to me. But I'm happy to discuss, maybe there is a better approach 🙌